### PR TITLE
Frustum Space Partitioning

### PR DESCRIPTION
**Pull request by OriolSors**

### CODE REVIEW

**Review by Dedado3000 — APPROVED** [review]
Good implementation, let's complete in the next sprint

File: Source/Quadtree.cpp [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	quadrantBoundingBox.SetFromCenterAndSize(newCenterFrontLeft, newSize);
+	frontLeftNode = new Quadtree(quadrantBoundingBox);
+
+	quadrantBoundingBox.SetFromCenterAndSize(newCenterBackRight, newSize);
+	backRightNode = new Quadtree(quadrantBoundingBox);
+
+	quadrantBoundingBox.SetFromCenterAndSize(newCenterBackLeft, newSize);
+	backLeftNode = new Quadtree(quadrantBoundingBox);
+
+	// Redistribution gameObjects part
```
Comment (marcelinus99) [inline Source/Quadtree.cpp:103]:
```suggestion
	// GameObject redistribution part
```

File: Source/Quadtree.cpp [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	for (std::list<GameObject*>::iterator it = gameObjects.begin(); it != gameObjects.end();)
+	{
+		GameObject* go = *it;
+
+		bool inFrontRight = frontRightNode->InQuadrant(go);
+		bool inFrontLeft = frontLeftNode->InQuadrant(go);
+		bool inBackRight = backRightNode->InQuadrant(go);
+		bool inBackLeft = backLeftNode->InQuadrant(go);
+
+		if (inFrontRight && inFrontLeft && inBackRight && inBackLeft) ++it;
```
Comment (xavi1111) [inline Source/Quadtree.cpp:115]: We will have to look closely if saving GameObjects on parent nodes is worth it.

File: Source/Quadtree.cpp [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+			if (inFrontLeft) frontLeftNode->Add(go);
+			if (inBackRight) backRightNode->Add(go);
+			if (inBackLeft) backLeftNode->Add(go);
+		}
+	}
+}
+
+void Quadtree::Clear()
+{
+	//TODO
```
Comment (marcelinus99) [inline Source/Quadtree.cpp:129]:
```suggestion
	// TODO: implement tree cleaning function
```

**Review by xavi1111 — DISMISSED** [review]
Good basic structure for quad tree

**Review by xavi1111 — APPROVED** [review]
Minor changes approved

**Review by Dedado3000 — COMMENTED** [review]
Seriously I need to confirm the commit to a comment

**Review by marcelinus99 — APPROVED** [review]
Good job!!

**Pull request merged**